cmd/opentelemetry/app/internal/esclient/client.go

```diff
@@ -172,20 +175,20 @@ func NewElasticsearchClient(params config.Configuration, logger *zap.Logger) (El
 	if err != nil {
 		return nil, err
 	}
-	if params.GetVersion() == 0 {
+	esVersion := int(params.GetVersion())
+	if esVersion == 0 {
 		esPing := elasticsearchPing{
 			username:     params.Username,
 			password:     params.Password,
 			roundTripper: roundTripper,
 		}
-		esVersion, err := esPing.getVersion(params.Servers[0])
+		esVersion, err = esPing.getVersion(params.Servers[0])
 		if err != nil {
 			return nil, err
 		}
 		logger.Info("Elasticsearch detected", zap.Int("version", esVersion))
-		params.Version = uint(esVersion)
```

Comment:
The ES version is derived here in the client factory method and assigned to the config. However the config is passed by value, hence the changes are not reflected after this function. Then the wrong value of the version was used by the called to create index template.

Instead of overriding the config it is cleaner to expose the version on the client.